cmd/oras/repository/tags.go

```diff
@@ -73,19 +79,42 @@ Example - Show tags of the target OCI layout archive 'layout.tar':
 }
 
 func showTags(opts showTagsOptions) error {
-	ctx, _ := opts.SetLoggerLevel()
+	ctx, logger := opts.SetLoggerLevel()
 	finder, err := opts.NewReadonlyTarget(ctx, opts.Common)
 	if err != nil {
 		return err
 	}
+	filter := ""
 	if opts.Reference != "" {
-		return fmt.Errorf("unexpected tag or digest %q found in repository reference %q", opts.Reference, opts.RawReference)
+		_, err := digest.Parse(opts.Reference)
+		if err == nil {
+			filter = opts.Reference
+		} else {
+			desc, err := finder.Resolve(ctx, opts.Reference)
+			if err != nil {
+				return err
+			}
+			filter = desc.Digest.String()
+		}
+		logger.Infof("[Preview] Querying tags associated to %s, it may take a while.\n", filter)
 	}
 	return finder.Tags(ctx, opts.last, func(tags []string) error {
 		for _, tag := range tags {
 			if opts.excludeDigestTag && isDigestTag(tag) {
 				continue
 			}
+			if filter != "" {
+				desc, err := finder.Resolve(ctx, tag)
+				if err != nil {
+					return err
+				}
+				if desc.Digest.String() != filter {
+					continue
+				}
+				if tag == opts.Reference {
+					continue
+				}
```

Comment:
Shouldn't the `oras repo tags` prints the input tag?

Reply:
I was thinking it would be more convenient for the user to not print that to save them the `grep -v latest`. They know that. I could be convinced otherwise though.

Reply:
I think this is about the completion of this functionality. That is, "show all tags of this manifest" versus "show all _other_ tags of this manifest". I think the former is less complex for new users and reduces chances to have bugs in scripting. Otherwise, we need to do something like
```bash
image="docker.io/library/alpine:latest"
tags=$(oras repo tags $image)
tags="$tags ${image#*:}"
```

Reply:
Another benefit to show all tags is that the output will be consistent for both inputs of digest and tag.